pkg/plugins/manager/loader/loader.go

```diff
@@ -199,7 +195,7 @@ func (l *Loader) Unload(ctx context.Context, pluginID string) error {
 		return plugins.ErrPluginNotInstalled
 	}
 
-	if !plugin.IsExternalPlugin() {
+	if plugin.IsCorePlugin() || plugin.IsBundledPlugin() {
```

Comment:
Note: This is changed to be more explicit in case a new plugin class is introduced where we don't want this error to be returned